Semantic Search Triggers (#18969)
blakeblackshear/frigate: docs/docs/configuration/semantic_search.md

4. Make your search language and tone closely match exactly what you're looking for. If you are using thumbnail search, **phrase your query as an image caption**. Searching for "red car" may not work as well as "red sedan driving down a residential street on a sunny day".
5. Semantic search on thumbnails tends to return better results when matching large subjects that take up most of the frame. Small things like "cat" tend to not work well.
6. Experiment! Find a tracked object you want to test and start typing keywords and phrases to see what works for you.

## Triggers

> Triggers in Frigate enable automated actions based on semantic search results for tracked objects. By defining triggers, you can configure Frigate to execute specific actions when a tracked object's image or description matches a predefined image or text, based on a similarity threshold. Triggers are managed per camera and can be configured via the Frigate UI in the Settings page under the Triggers tab.

Reviewer: ```suggestion
Triggers utilize semantic search to automate actions when a tracked object matches a specified image or description. Triggers can be configured so that Frigate executes a specific actions when a tracked object's image or description matches a predefined image or text, based on a similarity threshold. Triggers are managed per camera and can be configured via the Frigate UI in the Settings page under the Triggers tab.
```

### Configuration

Triggers are defined within the `semantic_search` configuration for each camera in your Frigate configuration file or through the UI. Each trigger consists of a `type` (either `thumbnail` or `description`), a `data` field (the reference image event ID or text), a `threshold` for similarity matching, and a list of `actions` to perform when the trigger fires.

#### Example Configuration

You can configure triggers directly in your config file, but this is more easily managed through the Frigate UI.

```yaml
cameras:
  front_door:
    semantic_search:
      triggers:
        red_car:
          enabled: true
          type: thumbnail
          data: 1751565549.853251-b69j73
          threshold: 0.7
          actions:
            - notification
        person_at_front_door:
          enabled: true
          type: description
          data: "Person in a red shirt standing at the front door"
          threshold: 0.8
          actions:
            - notification
```

> Fields:

Reviewer: I know we did this for the face / lpr, but as we move towards using the UI for the config I think it is best to leave this information to the reference config and maybe just link to that from here, vs redefining the entire config structure and having two places to need to update. (meaning we probably also remove it from the face / lpr config in 0.17)

Author: Yes, agreed. Will tweak.
